api/src/main/java/org/pmiops/workbench/cohortreview/CohortReviewServiceImpl.java

```diff
@@ -110,86 +105,118 @@ public DbCohortReview findCohortReview(Long cohortId, Long cdrVersionId) {
-              "Not Found: Cohort Review does not exist for cohortReviewId: %s", cohortReviewId));
-    }
-    return cohortReview;
+    return cohortReviewMapper.dbModelToClient(cohortReview);
   }
 
   @Override
-  public DbCohortReview findCohortReview(String ns, String firecloudName, Long cohortReviewId) {
-    DbCohortReview cohortReview =
-        cohortReviewDao.findByNamespaceAndFirecloudNameAndCohortReviewId(
-            ns, firecloudName, cohortReviewId);
-
-    if (cohortReview == null) {
-      throw new NotFoundException(
-          String.format(
-              "Not Found: Cohort Review does not exist for namespace: %s, firecloudName: %s and cohortReviewId: %d",
-              ns, firecloudName, cohortReviewId));
-    }
-    return cohortReview;
+  public CohortReview findCohortReview(Long cohortReviewId) {
+    DbCohortReview cohortReview = findDbCohortReview(cohortReviewId);
+    return cohortReviewMapper.dbModelToClient(cohortReview);
   }
 
   @Override
-  public void deleteCohortReview(DbCohortReview cohortReview) {
-    cohortReviewDao.delete(cohortReview);
+  public void deleteCohortReview(Long cohortReviewId) {
+    cohortReviewDao.delete(findDbCohortReview(cohortReviewId));
   }
 
   @Override
-  public List<DbCohortReview> getRequiredWithCohortReviews(String ns, String firecloudName) {
-    return cohortReviewDao.findByFirecloudNameAndActiveStatus(
-        ns,
-        firecloudName,
-        DbStorageEnums.workspaceActiveStatusToStorage(WorkspaceActiveStatus.ACTIVE));
+  public List<CohortReview> getRequiredWithCohortReviews(String ns, String firecloudName) {
+    return cohortReviewDao
+        .findByFirecloudNameAndActiveStatus(
+            ns,
+            firecloudName,
+            DbStorageEnums.workspaceActiveStatusToStorage(WorkspaceActiveStatus.ACTIVE))
+        .stream()
+        .map(cohortReviewMapper::dbModelToClient)
+        .collect(Collectors.toList());
   }
 
   @Override
-  public DbCohortReview saveCohortReview(DbCohortReview cohortReview) {
-    return cohortReviewDao.save(cohortReview);
+  public CohortReview saveCohortReview(CohortReview cohortReview, DbUser creator) {
+    return cohortReviewMapper.dbModelToClient(
+        cohortReviewDao.save(cohortReviewMapper.clientToDbModel(cohortReview, creator)));
   }
 
   @Override
   @Transactional
   public void saveFullCohortReview(
-      DbCohortReview cohortReview, List<DbParticipantCohortStatus> participantCohortStatuses) {
-    saveCohortReview(cohortReview);
+      CohortReview cohortReview, List<DbParticipantCohortStatus> participantCohortStatuses) {
+    cohortReviewDao.save(cohortReviewMapper.clientToDbModel(cohortReview));
     participantCohortStatusDao.saveParticipantCohortStatusesCustom(participantCohortStatuses);
   }
 
+  public CohortReview updateCohortReview(
+      CohortReview cohortReview, Long cohortReviewId, Timestamp lastModified) {
+    DbCohortReview dbCohortReview = findDbCohortReview(cohortReviewId);
```

Comment:
Should we wrap this around a try catch just in case cohortReviewId does not exist in DB

Reply:
I take it back i see findDbCohortReview is already throwing that execption

Reply:
Nvm i see findDbCohortReview is already doing that